src/themes/teams/components/Label/labelStyles.ts

```diff
@@ -1,42 +1,61 @@
+import * as _ from 'lodash'
+
 import { pxToRem } from '../../../../lib'
 import { ComponentSlotStylesInput, ICSSInJSStyle } from '../../../types'
 import { LabelProps } from '../../../../components/Label/Label'
+import { LabelVariables } from './labelVariables'
 
-const labelStyles: ComponentSlotStylesInput<LabelProps, any> = {
-  root: ({ props: { image, imagePosition, circular }, variables }): ICSSInJSStyle => ({
-    padding: variables.padding,
-    ...(image &&
-      imagePosition === 'start' && {
-        paddingLeft: variables.startPaddingLeft,
-      }),
-    ...(image &&
-      imagePosition === 'end' && {
-        paddingRight: variables.endPaddingRight,
-      }),
-    display: 'inline-flex',
-    alignItems: 'center',
-    height: variables.height,
-    fontSize: pxToRem(14),
-    lineHeight: variables.height,
-    backgroundColor: variables.backgroundColor,
-    color: variables.color,
-    borderRadius: pxToRem(3),
-    ...(circular && {
-      borderRadius: variables.circularRadius,
-    }),
-    overflow: 'hidden',
-  }),
-  image: ({ variables }): ICSSInJSStyle => ({
-    height: variables.height,
-    width: variables.height,
-  }),
-  icon: ({ props }): ICSSInJSStyle => ({
-    ...(props.icon &&
-      typeof props.icon === 'object' &&
-      (props.icon as any).onClick && {
-        cursor: 'pointer',
+const getColorSchemeFromObject = (colorScheme, colors) => {
```

Comment:
nit, you can use `_.mapValues` here:

```typescript
const getColorSchemeFromObject = (colorScheme, colors) => _.mapValues(colors, (color, colorName) => {
  // if the color scheme contains the color, then get the value from it, otherwise return the color provided
  const colorSchemeValue = _.get(colorScheme, color, colorScheme.default[color])
  return colorSchemeValue ? colorSchemeValue[colorName] : colors[colorName]
}
```